_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp

```diff
@@ -327,7 +329,14 @@ namespace UMC_MPEG2_DECODER
     // Find units in new bitstream buffer and process them
     UMC::Status MPEG2Decoder::AddOneFrame(UMC::MediaData * source)
     {
-        // The main parsing loop
+        if (m_lastSlice)
+        {
+            UMC::Status sts = AddSlice(m_lastSlice.release());
```

Comment:
are you sure, didn't you mean ```.get()```? You are loosing ownership of pointer and you need a guarantee that somebody will delete the owned memory inside.

Reply:
I'm 99 % sure:)

Reply:
ok, I just mention things that looks suspicious :)